contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -5864,3 +5870,48 @@ is_bbf_db_ddladmin_operation(Oid namespaceId)
 
 	return false;
 }
+
+/*
+ * remove_db_name_in_schema - remove the db name and underscore at the beginning
+ * 	of the given string. It is used to unmap schema name in error messages.
+ *
+ * 	@param schema_name - char *
+ * 	@return - unmapped schema name char *
+ */
+static const char *
+remove_db_name_in_schema(const char *object_name)
+{
+	char		*cur_db_name;
+	char		**splited_object_name;
+	char		*schema_name = NULL;
+	char		*mutable_name;
+    size_t		db_name_len;
+    size_t		prefix_len;
+	size_t		schema_name_len;
+
+	mutable_name = pstrdup(object_name);
+	splited_object_name = split_object_name(mutable_name);
+
+	// If there are more than two parts, then it’s a cross-db object name (db.schema.object) so we don’t need to deal with it
+	if (strlen(splited_object_name[1]) == 0)
+	{
+		schema_name = strlen(splited_object_name[2]) == 0 ? splited_object_name[3] : splited_object_name[2];
+		cur_db_name = get_cur_db_name();
+		db_name_len = strlen(cur_db_name);
+		schema_name_len = strlen(schema_name);
+		prefix_len = db_name_len + 1;
+
+		if (schema_name != NULL && schema_name_len > db_name_len && strncmp(schema_name, cur_db_name, db_name_len) == 0 && schema_name[db_name_len] == '_') {
+			// Return the part after the prefix
+			object_name += prefix_len;
```

Comment:
Instead of modifying the original string, I think it is better to return a new string. Here we can return the copy of source string starting from prefix_len